connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -485,16 +486,37 @@ public static synchronized void sdkReinitializeMobileConnect() {
         profile.deInitialize();
     }
 
-    public static ConnectivityManager getConnectivityManager() {
-        return connectivityManager;
+    /**
+     * Initialize components common to both Mobile Connect and ConnectID SDK profiles
+     */
+    public static synchronized void initializeCommonComponents() {
+        connectivityManager
+                = (ConnectivityManager) getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+        initalizeCellularNetwork();
+        initalizeDefaultNetwork();
     }
 
-    public static Network getCellularNetwork() {
-        return cellularNetwork;
+    public static boolean isCellularDataNetworkConnected() {
+        Validator.sdkInitialized();
+        NetworkInfo networkInfo = null;
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
+        } else {
+            if (cellularNetwork == null) {
+                return false;
+            }
+            networkInfo = connectivityManager.getNetworkInfo(cellularNetwork);
+        }
+        return (networkInfo != null) && networkInfo.isConnected();
     }
 
-    public static Network getWifiNetwork() {
-        return wifiNetwork;
+    public static boolean isCellularDataNetworkDefault() {
+        Validator.sdkInitialized();
```

Comment:
you should check for connectivityManager != null here rather than Validator.sdkInitialized